orangecontrib/prototypes/widgets/owfactoranalysis.py

```diff
@@ -70,38 +141,47 @@ def __init__(self): # __init__ je konstruktor (prva metoda ki se klice, ko se us
         table.horizontalHeader().hide()
         table.setShowGrid(False)
         box.layout().addWidget(table)
-
+        """
 
     @Inputs.data
     def set_data(self, dataset):
-        self.closeContext()
+        # self.closeContext()
         if dataset is None:
             self.sample = None
         else:
-            self.openContext(dataset.domain)  #Kaj je funkcija contexta
+            # self.openContext(dataset.domain)  # what is the function of context?
+            pass
 
         self.dataset = dataset
-        self.optionsBox.setDisabled(False)
-        self.commit.now() # Takoj poklici metodo commit
+        self.commit.now()
 
     def factor_analysis(self):
-        # Z izbranimi n_componentami izracunaj FA na self.dataset
-        result = FactorAnalysis(self.n_components).fit(self.dataset.X)
-        # Iz spremenljivke result (ki je instance nekega razreda) izlusci samo tabelo ki nas zanima
-        self.components = result.components_
+        # with chosen n_components and depending on the user-selected rotation, calculate the FA on self.dataset
+        if self.setting_for_rotation == 0:
+            result = FactorAnalysis(self.n_components).fit(self.dataset.X)
+        elif self.setting_for_rotation == 1:
+            result = FactorAnalysis(self.n_components, rotation="varimax").fit(self.dataset.X)
+        elif self.setting_for_rotation == 2:
+            result = FactorAnalysis(self.n_components, rotation="quartimax").fit(self.dataset.X)
+        else:
+            print("Error:")
+
+        # from result variable (instance of class) only extract the table we are interested in (components)
+        calculated_components = result.components_
 
-        # Pretvori tabelo nazaj v Orange.data.Table
+        # transform the table back to Orange.data.Table
         self.result = Table.from_numpy(Domain(self.dataset.domain.attributes),
```

Comment:
If you use `self.dataset.domain.attributes` multiple times, create a variable

```python
attributes = self.dataset.domain.attributes
```

Variables are cheap.

Reply:
As said, `result` is not a good name. Use `factors` or `loadings` or ... whatever the result is.

Also, you need to have `self.result = None` in `__init__`. `__init__` should always initialize all attributes.